internal/config/config.go

```diff
@@ -434,6 +443,11 @@ func New(opts ...ConfigOption) (*Config, error) {
 		return nil, xerrors.Errorf("failed to merge in %v config: %w", envSecrets, err)
 	}
 
+	// Merge in config defined with environment variables
+	if err := mergeInEnvConfig(v); err != nil {
```

Comment:
Env var should be setup via viper, https://github.com/spf13/viper#what-is-viper. why do we need to merge it explicitly?

Reply:
This is specifically for CLIENT_CONFIG_OVERRIDE, which is input as a Json string and we need to unmarshal this string into the client config.

Reply:
Sync offline, indeed viper can handle all environment variable Unmarshaling to the config, so will use that instead.